### DISCUSSION TITLE
DWMBG Config Site

### DISCUSSION THREAD

**Discussion by Sand216** [Maplespe/DWMBlurGlass#306] [General]
Hey! I have a hobby of web design and had an idea to make a website to easily store and showcase config files for DWMBlurGlass. This can be used to get different Windows 7 colors without having to mess with sliders and color values in the gui, for example. This is purely for sharpening my skills (and fun) but I wanted to get confirmation if this is okay with you guys. If you do not want me hosting the site or even creating it at all please let me know. It may never even come to fruition, but if I do figure out how to make everything work I wanted to make sure it is in your best interest.

Here is a snippet of what I have demoed so far, I am surprised with myself on how far I have come with web development :)

(Ugly black and white gradient is for demonstration of the "glass" effects)

https://github.com/Maplespe/DWMBlurGlass/assets/164842290/721b7ae8-834b-46ed-b0bc-540fdfe86eff

**Comment by Maplespe** [comment]
I think it's fine. Why not?

**Comment by TheInsane101** [comment]
This is awesome! And if not, I'll just make an r/WindowsAero subreddit.

I've been collecting and putting together the hex and ColorizationBalance values for all default Windows colours, from Windows 7 Milestone 3 6801 all the way up to 8.1 colours. Can my collection become the first pinned post if this website of yours comes to fruition?

[Windows 7 M3 - 8.1 default colours & theme colours collection.zip](https://github.com/Maplespe/DWMBlurGlass/files/15211009/Windows.7.M3.-.8.1.default.colours.theme.colours.collection.zip)

**Reply by Sand216** [reply]
This is quite the collection! I will for sure add these.

Also, I should probably clarify what I plan to do for this project. There would be no "posts" per say, mostly everything would be handled on Github. I am planning to make it similar to the [Firefox CSS Store](https://github.com/FirefoxCSS-Store/FirefoxCSS-Store.github.io), where new configs can be added via pull request or issue. Config names, descriptions, authors, and image links would be stored in a json file. I have never worked with json files before, so I have my work cut out for me 😅.

**Reply by TheInsane101** [reply]
Oh cool! I was imagining more of a forum kind of website but your idea is possibly better than I expected! Would be nice to have a part where users could hang out and discuss like over here, though that may not be necessary since we DO have here.

Btw, the pack above has some mistakes. I've just edited the above comment with accuracy improvements. Could you please redownload it? Sorry🙈

I may make the Vista presets later on and post that in another reply. 

Good luck with the journey of creating the site!

**Reply by Sand216** [reply]
@TheInsane101 I have been taking a closer look at these configs and there are some things I am not understanding. 

For many of the configs, `useAccentColor=true` is set. This overrides the colors with the system's accent color, which I would assume is unintended behavior.

All the configs have `blurAmount=3` and `customBlurAmount=3` which leads to there being almost no blur at all, at least on Windows 11.

**Screenshot showing the system accent color in place of the Sky color and there being almost no blur at all.**

![Screenshot 2024-05-27 115050](https://github.com/Maplespe/DWMBlurGlass/assets/164842290/9b54348d-5cbc-4bb6-9b50-78b5b02d579b)

Many times there are colors set at 0 opacity, but it doesn't seem to affect the effect. It may be confusing for some to see a different color in the preview than the actual titlebars, though.

As far as I know, Windows 7 always had black text in the titlebars, but for dark mode it is set to white.

**Screenshot showcasing the two previous statements.**

![Screenshot 2024-05-27 114135](https://github.com/Maplespe/DWMBlurGlass/assets/164842290/7f256a7a-54ee-4643-8361-c6e5957e9e2a)

I will probably uncheck `Override AccentBlur Effect` in these configs as it's use is very situational and can have incompatibilities. 

These configs also seem to be very Windows 10 focused, is this the case? I am lead to believe this as the configs have `extendBorder=true` and `applyglobal=false` (The DWMAPI Override). I want to know what to set the configs' type as to avoid incompatibilities, and possibly make versions better compatible with Windows 11.

Thanks for putting together these configs, it will make a great starting library!

**Reply by TheInsane101** [reply]
OK, thank you for explaining the issues! I did indeed make these for Windows 10, so make sure to tag them with that, and you can optimise/convert them to be compatible with Windows 11. Could you also explain to me how to make future configs like this too? I'll then post duplicates of future configs: one for Windows 10 and one for Windows 11, unless DWMBG gets other updates ofc.

I intentionally set overrideaccent to true, but realise that that may have been a bad decision. You can revert if it you feel that it may cause more issues than benefits. 

I didn't touch the dark mode settings, because I'm using msstyles themes and 7tsp icon packs, the looks of which get messed up when enabling dark mode. You can edit/optimise these to work on dark mode for me too, thanks!

By the way, I have discovered that Windows 7's 20 regional themes each had TWO versions: one that came with the different Windows 7 ISOs, and the other which could be downloaded from Microsoft's website. I am currently working on porting over the themes I have missed and will give you the updated, more accurate collection later!

**Reply by Sand216** [reply]
> OK, thank you for explaining the issues! I did indeed make these for Windows 10, so make sure to tag them with that, and you can optimise/convert them to be compatible with Windows 11. Could you also explain to me how to make future configs like this too? I'll then post duplicates of future configs: one for Windows 10 and one for Windows 11, unless DWMBG gets other updates ofc.

Really the only thing you have to do to make a config "Windows 11 compatible" is to uncheck extend to borders and check override DWMAPI. I am unsure if the lack of blur caused by the global and titlebar radius both being set to three is a Windows 11 exclusive issue, but I normally just set the titlebar radius and leave global unchecked.

> By the way, I have discovered that Windows 7's 20 regional themes each had TWO versions: one that came with the different Windows 7 ISOs, and the other which could be downloaded from Microsoft's website. I am currently working on porting over the themes I have missed and will give you the updated, more accurate collection later!

Great, thanks!

**Reply by Sand216** [reply]
As I do not have access to a Windows 10 machine, could you take a screenshot for the preview section for each config? You can keep your custom msstyles on if you want if it a Windows 7 one. Try to keep the screenshots around a 16:9-ish aspect ratio to make sure it looks good in the space allocated.

Here are some of the updated configs with dark mode support and some of the issues fixed, if it helps at all. I have not made Windows 11 versions yet.

[Win7ConfigsWin10.zip](https://github.com/Maplespe/DWMBlurGlass/files/15463415/Win7ConfigsWin10.zip)

**Comment by Sand216** [comment]
I may be able to post some screenshots of early designs to get some feedback soon. I should be able to speed up development when my schedule frees up.

**Comment by Sand216** [comment]
A couple weeks overdue, but I finally have stuff to share! I have shifted the style of the site away from the Aero Glass look, as DWMBG isn't just for Aero Glass but for all sorts of blurs and effects, and I want my site to have the same appeal. I ended up deciding to recreate MiaoUi in HTML and CSS, in what I have been internally calling MeowUI (I saw a pun opportunity, I had to go for it, sorry 😅).

Anyways, here are some screenshots!
### A video displaying the site's new look, along with the design for the config panels.
I don't know what to put for a background image, let me know any ideas.

https://github.com/Maplespe/DWMBlurGlass/assets/164842290/535d1015-744b-4d5e-a7de-8b60c6b788fe
### A screenshot of the current, temporary config files in JSON format.
![Screenshot 2024-05-25 223655](https://github.com/Maplespe/DWMBlurGlass/assets/164842290/2979967e-1b7f-4f29-b0ac-b04800bf0d6a)
### A screenshot of an example config in JSON format.
![Screenshot 2024-05-25 224144](https://github.com/Maplespe/DWMBlurGlass/assets/164842290/1d00a4a8-4cea-4671-988a-7eacf5e38123)

Also, it should be fairly simple for me to add new tags, so if you have any ideas for any please let me know!

**Reply by TheInsane101** [reply]
Looks awesome! I can't wait to see the final version!

**Comment by Sand216** [comment]
It’s been quite a while since I’ve last touched this, it’s not canceled, I’ve just not found the time to work on it :P

There isn’t too much more for me to finish once I pick it back up.
